PeerioTechnologies/peerio-mobile
# feat: welcome back screen [branch ch12939]

#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/12939/mobile-welcome-back-screen
#### Testing instructions  
Entering the app after signing out from an account should show the "Welcome Back" screen, with the username and firstname of the last account. You can press switch user to show the regular sign in screen without any saved info, or go back to 'Welcome" screen (the one with the illustration) by pressing the back arrow button. Either one of those actions clears the last user, so restarting the app again will take the user to the "Welcome screen" with the illustration.

Here are all the possible combinations of scenarios and their expected results:

1- Opening app "fresh" (for the first time, or without having previously logged in)
Expected: See Welcome screen (the one with the illustration)

2- Sign in -> refresh app
Expected: User is signed in

3- Sign in -> logout -> app restarts -> open app
Expected: See Welcome Back Screen (Welcome back, *Name*)

4- Repeat step 3 -> after app is opened and welcome back screen is visible -> refresh app
Expected: See Welcome Back Screen (Welcome back, *Name*)

5- Repeat step 3 -> after app is opened and welcome back screen is visible -> press back button
Expected: See Welcome Screen (illustration)

6- Repeat step 5 -> after seeing Welcome Screen -> refresh
Expected: Should see Welcome Screen again

7- Repeat step 3 -> after app is opened and welcome back screen is visible -> press "change user"
Expected: See Welcome Back Screen (Welcome back, NO name)

7- Repeat step 7 -> refresh app
Expected: See Welcome Screen (illustration)

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
